homeassistant/components/ecobee/humidifier.py

```diff
@@ -137,12 +137,16 @@ def set_mode(self, mode):
             )
 
         self.data.ecobee.set_humidifier_mode(self.thermostat_index, mode)
-        self.update_without_throttle = True
+
+        # set_humidifier_mode was successful, reflect changes in thermostat
+        self.thermostat["settings"]["humidifierMode"] = mode
 
     def set_humidity(self, humidity):
         """Set the humidity level."""
         self.data.ecobee.set_humidity(self.thermostat_index, humidity)
-        self.update_without_throttle = True
+
+        # set_humidity was successful, reflect changes in thermostat
+        self.thermostat["runtime"]["desiredHumidity"] = humidity
```

Comment:
If you want to write the state optimistically, you'll need to call `self.schedule_update_ha_state()` or the entity won't be updated until the next poll

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L635